hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java

```diff
@@ -28,19 +26,20 @@
  */
 public abstract class HdfsNode implements IOfferEvaluator, ILauncher {
   private final Log log = LogFactory.getLog(HdfsNode.class);
-  private final LiveState liveState;
   private final ResourceFactory resourceFactory;
 
   protected final HdfsFrameworkConfig config;
-  protected final IPersistentStateStore persistenceStore;
+  protected final HdfsState state;
```

Comment:
I gave it a try turning HdfsState into an interface and injecting it instead.  It turns into a mess.

Essentially the only thing different between the two proposed implementations of HdfsState will be the StateFactory they use.  So to avoid code duplication you end up having an internal implementation object which takes a StateFactory.  This object is exactly what HdfsState is right now.

Then you have the to HdfsState interface implementations pass in either an InMemory or a ZKStateFactory.  But in order to avoid all the code duplication required to pass all call HdfsState interface calls to the internal implementation, you have them extend from an Abstract parent class.  Jeez.  This is not better.

We inject the persistence layer into the HdfsState object now.  It's one line in an injection module and we're done.